src/cmd/flash.js

```diff
@@ -55,11 +59,108 @@ module.exports = class FlashCommand extends CLICommandBase {
 		} else if (local) {
 			let allFiles = binary ? [binary, ...files] : files;
 			await this.flashLocal({ files: allFiles, applicationOnly, target });
+		} else if (tachyon) {
+			let allFiles = binary ? [binary, ...files] : files;
+			await this.flashTachyon({ verbose, files: allFiles });
 		} else {
 			await this.flashCloud({ device, files, target });
 		}
 	}
 
+	async flashTachyon({ verbose, files }) {
+		this.ui.write(`${os.EOL}Ensure only one device is connected to the computer${os.EOL}`);
+
+		let unpackToolFolder;
+		if (files.length === 0) {
+			// If no files are passed, assume the current directory
+			unpackToolFolder = process.cwd();
+			files = await fs.readdir(unpackToolFolder);
+		} else if (files.length === 1) {
+			// If only one file is passed, check if it's a directory
+			const stats = await fs.stat(files[0]);
+			if (stats.isDirectory()) {
+				unpackToolFolder = files[0];
+				files = await fs.readdir(files[0]);
+			}
+		} else {
+			// If multiple files are passed, check the directory from the first file
+			unpackToolFolder = path.dirname(files[0]);
+		}
+
+		let linxuFiles = await this._findLinuxExecutableFiles(files, { directory: unpackToolFolder });
+		linxuFiles = linxuFiles.map(f => path.basename(f));
+
+		const elfFiles = linxuFiles.filter(f => f.startsWith('prog') && f.endsWith('.elf'));
+		const rawProgramFiles = linxuFiles.filter(f => f.startsWith('rawprogram') && f.endsWith('.xml'));
+		const patchFiles = linxuFiles.filter(f => f.startsWith('patch') && f.endsWith('.xml'));
+
+		if (!elfFiles.length || !rawProgramFiles.length) {
+			throw new Error('The directory should contain at least one .elf file and one rawprogram file');
+		}
+
+		const sortByNumber = (a, b) => {
+			const extractNumber = str => parseInt(str.match(/(\d+).xml/)[1]);
+			return extractNumber(a) - extractNumber(b);
+		};
+
+		rawProgramFiles.sort(sortByNumber);
+		patchFiles.sort(sortByNumber);
+
+		if (rawProgramFiles.length !== patchFiles.length) {
+			throw new Error('The number of rawprogram files should match the number of patch files');
+		}
+
+		let filesToProgram = [];
+		// interleave the rawprogram files and patch files
+		for (let i = 0; i < rawProgramFiles.length; i++) {
+			filesToProgram.push(rawProgramFiles[i]);
+			filesToProgram.push(patchFiles[i]);
+		}
+
+		filesToProgram.unshift(elfFiles[0]);
+
+		this.ui.write(`Found the following files in the directory:${os.EOL}`);
+		this.ui.write('  Loader file:');
+		this.ui.write(`    - ${elfFiles[0]}${os.EOL}`);
+
+		this.ui.write('  Program files:');
+		for (const file of rawProgramFiles) {
+			this.ui.write(`    - ${file}`);
+		}
+		this.ui.write(os.EOL);
+
+		this.ui.write('  Patch files:');
+		for (const file of patchFiles) {
+			this.ui.write(`    - ${file}`);
+		}
+		this.ui.write(os.EOL);
+		
+		const archType = utilities.getArchType();
+		const qdl = path.join(__dirname, `../../assets/qdl/${archType}/qdl`);
+		await fs.ensureFile(qdl);
```

Comment:
Another piece of feedback: extract the execution of qdl to a separate file `lib/qdl.js`. See the example of the old `dfu.js` https://github.com/particle-iot/particle-cli/blob/v1.43.2/src/lib/dfu.js

Keep using the new execa, but put stuff like getting the correct executable for the current OS and platform in a helper function in `lib/qdl.js` so that `flash.js` just needs to call a function to perform the qdl flash.